lib/Module.js

```diff
@@ -575,6 +576,7 @@ class Module extends DependenciesBlock {
 					switch (/** @type {BuildMeta} */ (this.buildMeta).defaultObject) {
 						case "redirect":
 						case "redirect-warn":
+						case "redirect-warn-only":
```

Comment:
Honestly I think current `redirect-warn` actually means `redirect-error`, because if parent is `strict`, `redirect-warn` do thing but warn, but I don't to break current users, so I added a new `redirect-warn-only` which is equal to `redirect`, just add warning if user use named exports